task16/task16/Quaternion.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace task16
+{
+    public struct Quaternion
+    {
+        private readonly double _a;
+        private readonly double _b;
+        private readonly double _c;
+        private readonly double _d;
+
+        public double A => _a;
+        public double B => _b;
+        public double C => _c;
+        public double D => _d;
+
+        public double Abs => Math.Sqrt(_a * _a + _b * _b + _c * _c + _d * _d);
+
+        public Quaternion(double a, double b, double c, double d)
+        {
+            _a = a;
+            _b = b;
+            _c = c;
+            _d = d;
```

Comment:
В вашей разработке свойства A, B, C и D только для чтения. В задании такого не было.
```suggestion
        public double A { get; set; }
        public double B { get; set; }
        public double C { get; set; }
        public double D { get; set; };

        public double Abs => Math.Sqrt(_a * _a + _b * _b + _c * _c + _d * _d);

        public Quaternion(double a, double b, double c, double d)
        {
            A = a;
            B = b;
            C = c;
            D = d;
```
  Соответсвтенно, во всем оств=альном коде нужно заменить `_a` на `A` и т. п.